docs/on-demand-gather.yaml

```diff
@@ -0,0 +1,25 @@
+# Example of a DataGather resource that uses a PersistentVolume to store the data.
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: on-demand-gather-pvc
+  namespace: openshift-insights
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+  storageClassName: gp2-csi
+---
+apiVersion: insights.openshift.io/v1alpha1
+kind: DataGather
+metadata:
+  name: on-demand-gather-job
+spec:
+  storage:
+    type: PersistentVolume
+    mountPath: /data
+    persistentVolume:
+      claim:
+        name: on-demand-gather-pvc
```

Comment:
I think we can put this to the official OCP documentation but I would probably not keep it here. Just my 2 cents...

Reply:
I think it's nice to have it in a repository so anyone can use it to create the on-demand gathering job without going to the OCP docs. Otherwise, feel free to drop, but let's see what @ncaak thinks about it being here

Reply:
No strong opinions here. I think we could put this in the OCP documentation, for sure. But I don't see why it shouldn't be here as an example. Maybe with a more explanatory comment that gives some insight into what it is used for?
@tremes what's your concern to not have this example here?

Reply:
Not really a concern, but my take is that the PVC definition can be easily found in Kubernetes docs and in general it would be nice to have some context for those yaml files. For example something like if you want to do A then you need to configure B,C etc. 
Perhaps `docs/arch.md` could be extended.

Reply:
I will try to update `docs/arch.md`

Reply:
`docs/arch.md` is updated now, so you can review it